app/views.py

```diff
@@ -0,0 +1,242 @@
+                    "type":device.type,
+                    "user":device.user.id
+                }}, status=status.HTTP_200_OK)
+        except:
+            return Response({"message":"User does not have a registered device"}, status=status.HTTP_400_BAD_REQUEST)
+            
+# Send Alert notification to all the devices other than the users device
+class FCMPushNotificationView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        lat = request.data["latitude"]
+        lon = request.data["longitude"]
+        try:
+            # Checking if the user making quesry has a registered device
+            user = request.user
+            try:
+                device = FCMDevice.objects.get(user=user)
+            except:
+                return Response({"message":"The User's device is not registered"}, status=status.HTTP_400_BAD_REQUEST)
+            # To get all devices other than the one who made request
+            devices = FCMDevice.objects.exclude(user=user)
+            devices.send_message(data={"lat":lat, "lon":lon, "user_id":user.id})
+           
+           # Creating a new request for help in the database
+            req_ser = RequestsSerializer(data={
+                "user_id":user.id,
+                "latitude":lat,
+                "longitude":lon
+            })
+            if req_ser.is_valid():
+                req_ser.save()
+                return Response({"message":"Sent notificaion", "Request":req_ser.data}, status=status.HTTP_200_OK)
+            else:
+                return Response({"message":req_ser.errors}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            return Response({"message":str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ViewRequests(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        user = request.user
+        today_date = datetime.today().strftime('%Y-%m-%d')
+        req_objects = Requests.objects.filter(Q(date_time_creation__date=today_date) & ~Q(user_id=user.id))
+        response = RequestsSerializer(req_objects, many=True)
+        resp = response.data
+        for req in resp:
+            user_req = User.objects.get(id=req['user_id'])
+            req['user_username'] = user_req.username
+        return Response({"message":"Received Requests", "Requests":resp}, status=status.HTTP_200_OK)
+
+
+class ViewChatRooms(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        user = request.user
+        chatRooms = ChatRoom.objects.filter( Q(participant1_id=user.id) | Q(participant2_id=user.id))
+        chatRooms_serializer = ChatRoomSerializer(chatRooms, many=True)
+        resp = chatRooms_serializer.data
+        if len(resp) == 0:
+            return Response({"message":"No chat rooms available"}, status=status.HTTP_204_NO_CONTENT)
+        else:
+            resp = chatRooms_serializer.data
+            for chatroom in resp:
```

Comment:
why are you iterating over the chatrooms. There will be only one chatroom between two participant.

Reply:
There might be multiple chatrooms associated with a single person. So each person might be having conversations with multiple people.